frontend/src/metabase/embedding-sdk/components/SimpleDataPicker/SimpleDataPicker.tsx

```diff
@@ -1,80 +1,90 @@
-  options: Options[];
-  onClick: (option: any) => void;
-}
-
-interface Options {
-  id: number;
-  display_name: string;
+  isInitiallyOpen: boolean;
+  triggerElement: ReactNode;
+  setSourceTableFn: (tableId: TableId) => void;
 }
 
-const TEN_OPTIONS_HEIGHT = 10 * 33;
-
 export function SimpleDataPicker({
   selectedEntity,
-  options,
-  onClick,
+  isInitiallyOpen,
+  setSourceTableFn,
+  triggerElement,
 }: SimpleDataPickerProps) {
-  const shouldShowSearchBar = options.length > 10;
-  const [searchText, setSearchText] = useState("");
-  function filterSearch(option: Options): boolean {
-    if (searchText) {
-      return normalizeString(option.display_name).includes(
-        normalizeString(searchText),
-      );
+  const [isDataPickerOpened, { toggle, close }] =
+    useDisclosure(isInitiallyOpen);
+  const { data, isLoading, error } = useSearchQuery({
+    models: ["dataset", "table"],
+  });
+
+  const options = useMemo(() => {
+    if (!data) {
+      return [];
     }
 
-    return true;
-  }
+    return sortEntities(
+      data.data.map(entity => {
+        return {
+          ...entity,
+          id: isModel(entity)
+            ? getQuestionVirtualTableId(entity.id)
+            : entity.id,
+        };
+      }),
+      { sort_column: "name", sort_direction: SortDirection.Asc },
+    );
+  }, [data]);
 
   return (
-    <Paper w="300px" p="sm">
-      {shouldShowSearchBar ? (
-        <TextInput
-          data-autofocus
-          type="search"
-          icon={<Icon name="search" size={16} />}
-          mb="sm"
-          placeholder={t`Search…`}
-          onChange={e => setSearchText(e.target.value ?? "")}
-        />
-      ) : (
-        /**
-         * Behave like Mantine 7's `FocusTrap.InitialFocus`.
-         *
-         * This component disable the automatic focus on the first focusable element
-         * when there is no search bar.
-         */
-        <span aria-hidden data-autofocus tabIndex={-1} />
-      )}
-      {/* @ts-expect-error - I think the typing for ScrollArea.Autosize is wrong. This might be fixed in Mantine 7 */}
-      <ScrollArea.Autosize mah={TEN_OPTIONS_HEIGHT} type="auto">
-        {options.filter(filterSearch).map(option => {
-          return (
-            <NavLink
-              key={option.id}
-              active={selectedEntity === option.id}
-              icon={<Icon c="var(--mb-color-icon-primary)" name="table" />}
-              label={option.display_name}
-              onClick={() => {
-                onClick(option);
-              }}
-              variant="default"
-            />
-          );
-        })}
-      </ScrollArea.Autosize>
-    </Paper>
+    <Popover
+      opened={isDataPickerOpened}
+      position="bottom-start"
+      onClose={close}
+    >
+      <Popover.Target>
+        <Box onClick={toggle}>{triggerElement}</Box>
+      </Popover.Target>
+      <Popover.Dropdown>
+        <DelayedLoadingAndErrorWrapper loading={isLoading} error={error}>
```

Comment:
This is a newer loading indicator that is implemented to improve user performance perception. It will not flash the loading icon when the result is loaded super fast, and will only show it after a certain timeout.